server/odc-service/src/main/java/com/oceanbase/odc/service/git/repo/RepositoryManager.java

```diff
@@ -0,0 +1,212 @@
+import com.oceanbase.odc.service.git.GitClientOperator;
+import com.oceanbase.odc.service.git.GitIntegrationService;
+import com.oceanbase.odc.service.git.model.GitIntegrationProperties;
+import com.oceanbase.odc.service.git.model.RepoState;
+import com.oceanbase.odc.service.iam.auth.AuthenticationFacade;
+import com.oceanbase.odc.service.objectstorage.cloud.CloudObjectStorageService;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author: liuyizhuo.lyz
+ * @date: 2024/8/7
+ */
+@Slf4j
+@Component
+public class RepositoryManager {
+    private static final String LOCK_FORMAT = "repo-lock-%s-%s";
+    private static final String GIT_REPO_DIR =
+            Paths.get(MoreObjects.firstNonNull(SystemUtils.getEnvOrProperty("file.storage.dir"), "./data"), "repos")
+                    .toString();
+
+    @Autowired
+    private GitRepoRepository gitRepoRepository;
+    @Autowired
+    private GitRepoStageRepository stageRepository;
+    @Autowired
+    private EncryptionFacade encryptionFacade;
+    @Autowired
+    private AuthenticationFacade authenticationFacade;
+    @Autowired
+    private CloudObjectStorageService objectStorageService;
+    @Autowired
+    private JdbcLockRegistry jdbcLockRegistry;
+    @Autowired
+    private GitIntegrationService integrationService;
+
+    /**
+     * Use {@code repoId}、{@code userId} to identify repository
+     */
+    private final LoadingCache<Pair<Long, Long>, GitClientOperator> operatorCache;
+
+    public RepositoryManager(@Autowired GitIntegrationProperties gitIntegrationProperties) {
+        operatorCache = Caffeine.newBuilder()
+                .maximumSize(gitIntegrationProperties.getGitRepositoryMaxCachedSize())
+                .expireAfterAccess(Duration.ofMinutes(gitIntegrationProperties.getGitRepositoryPreserveMinutes()))
+                .removalListener((pair, operator, cause) -> {
+                    if (operator != null) {
+                        deleteRepo((GitClientOperator) operator);
+                        ((GitClientOperator) operator).close();
+                    }
+                })
+                .build(this::createGitOperator);
+    }
+
+    @PostConstruct
+    public void init() {
+        try {
+            FileUtils.cleanDirectory(new File(GIT_REPO_DIR));
+        } catch (Exception e) {
+            log.warn("failed to clean git repo directory {}", GIT_REPO_DIR);
+        }
+    }
+
+    public GitClientOperator getOperator(Long repoId, Long userId) {
+        return operatorCache.get(new Pair<>(repoId, userId));
+    }
+
+    private GitClientOperator createGitOperator(Pair<Long, Long> pair) throws IOException, GitAPIException {
+        File repo = getRepoDir(pair.left, pair.right);
+        FileUtils.forceMkdir(repo);
+        String[] children = repo.list();
+        if (children.length == 0) {
+            return initRepo(repo, pair.left, pair.right);
+        }
+        Verify.equals(1, children.length, "found more than one repositories");
```

Comment:
I modified this into : if the `children.lenth > 0` , we would clean up this directory.
Normally, it would not follow this logic. This is for safety reasons